### DISCUSSION TITLE
Discogs to allow option to append style to genre

### DISCUSSION THREAD

**Discussion by jdheyburn** [beetbox/beets#4301] [Ideas]
As mentioned before in various places ([example](https://discourse.beets.io/t/help-fetching-discogs-style-and-mapping-to-write-to-genre-field/1557/21)) the genre for Discogs items is too broad, and its usually in Style where the verbosity can be found.

Most players only support the tag GENRE and so when beets sets this Genre from discogs, it can be too vague when viewing it back. I.e. Electronic is a genre whereas Techno is a style, and it could be more useful to have the style appended to the genre so that in a player, it would display "Electronic; Techno" or something along those lines. I know in the beets db that style is already being set, but this isn't useful for playing back the media.

Does it make sense to add this configuration?

It would also be useful to allow beets to run across all items in the music library to change this, similar to how lastgenre can be used to change genres for all items:

```bash
beet lastgenre -a
```

**Comment by sampsyo** [comment]
Seems like a pretty straightforward feature to consider! I'm not a big Discogs user, but I can definitely see this being useful.

**Reply by jdheyburn** [reply]
Awesome! I don't mind getting a PR raised for it - I've been wanting to contribute back to beets since I started using it last year.

**Reply by jdheyburn** [reply]
@sampsyo [have created the PR](https://github.com/beetbox/beets/pull/4304), looking forward to the review! I didn't get the CLI tool to fix existing genres, I'll add that in a separate PR once this one has been given the thumbs up 👍🏻
